spring-jdbc/src/main/java/org/springframework/jdbc/datasource/DataSourceUtils.java

```diff
@@ -177,11 +177,11 @@ public static Integer prepareConnectionForTransaction(Connection con, @Nullable
 			throws SQLException {
 
 		Assert.notNull(con, "No Connection specified");
-
+		final boolean debugEnabled = logger.isDebugEnabled();
```

Comment:
We don't normally make local variables `final`.

```suggestion
		boolean debugEnabled = logger.isDebugEnabled();
```